src/wfs/src/main/java/org/geoserver/wfs/kvp/TypeNameKvpParser.java

```diff
@@ -43,8 +42,7 @@ protected Object parseToken(String token) throws Exception {
             if (ftInfo == null) {
                 return new QName(null, token);
             } else {
-                final Name name = ftInfo.getFeatureType().getName();
-                return new QName(name.getNamespaceURI(), name.getLocalPart());
+                return new QName(ftInfo.getNamespace().getURI(), token);
```

Comment:
I had to stop for a moment on this change, so indeed before we were always using the layer native name, this was indeed not allowing any type of aliasing. This change looks good to me and seems to be backward compatible, and the fact that tests passed makes believe that indeed this is full backward compatible.